megamek/src/megamek/client/bot/princess/InfantryFireControl.java

```diff
@@ -115,8 +115,10 @@ public double getMaxDamageAtRange(final MovePath shooterPath, final MovePath tar
 
             // case 1
             if (weaponType.hasFlag(WeaponType.F_INFANTRY)) {
+                int infantryCount = shooter.isConventionalInfantry() ?
+                        ((Infantry) shooter).getInternal(Infantry.LOC_INFANTRY) : 1;
```

Comment:
Cast to Infantry isn't needed here.

Reply:
It's used by the bot to calculate incoming and outgoing damage when evaluating a path being considered by an infantry unit. So it was trying to evaluate a move path for a foot platoon and checking how much damage an "apocalypse world rover" would do to the foot platoon. That particular tank has a weapon with an infantry flag, but was not an infantry unit. 

I'm pretty sure this will lead to incorrect behavior for battle armor, so I'm going to take another look at it.

Reply:
There, that should do it.